azkaban-common/src/main/java/azkaban/server/AzkabanServer.java

```diff
@@ -75,12 +77,27 @@ public static Props loadProps(String[] args, OptionParser parser) {
       azkabanSettings = loadConfigurationFromAzkabanHome();
     }
 
+    if (azkabanSettings != null) {
```

Comment:
This is a functional change in itself - perhaps the only one in the PR. But this is required for the refactor.

The existing code modifies the config while setting up the server. This creates a spaghetti flow and I had to disentangle this to make that flow somewhat linear. The null check is required else some test fails. I don't want to focus on that now. The current direction is to make the initialization simple so that I can introduce Guice at the root level. This would simplify refactoring drastically since a lot of the entanglement is due to static referencing of data/ members of one class from another. This will be evident in subsequent PRs